control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -337,28 +341,47 @@ func reconcileAWSEndpointService(ctx context.Context, awsEndpointService *hyperv
 		if !hasAWSConfig(&hcp.Spec.Platform) {
 			return fmt.Errorf("AWS platform information not provided in HostedControlPlane")
 		}
-		input := &ec2.CreateVpcEndpointInput{
-			ServiceName:     aws.String(serviceName),
-			VpcId:           aws.String(hcp.Spec.Platform.AWS.CloudProviderConfig.VPC),
-			VpcEndpointType: aws.String(ec2.VpcEndpointTypeInterface),
-			SubnetIds:       []*string{hcp.Spec.Platform.AWS.CloudProviderConfig.Subnet.ID},
-			TagSpecifications: []*ec2.TagSpecification{{
-				ResourceType: aws.String("vpc-endpoint"),
-				Tags:         apiTagToEC2Tag(hcp.Spec.Platform.AWS.ResourceTags),
-			}},
-		}
-		output, err := ec2Client.CreateVpcEndpointWithContext(ctx, input)
+
+		// Verify there is not already an Endpoint that we can adopt
+		// This can happen if we have a stale status on AWSEndpointService or encoutered
+		// an error updating the AWSEndpointService on the previous reconcile
+		output, err := ec2Client.DescribeVpcEndpointsWithContext(ctx, &ec2.DescribeVpcEndpointsInput{
+			Filters: apiTagToEC2Filter(awsEndpointService.Name, hcp.Spec.Platform.AWS.ResourceTags),
```

Comment:
Names are unique within a basedomain... is relying on name plus user-defined tags open to adoption bugs for clusters across domains?

Reply:
the infra id for the cluster must be unique and we are matching on a `{infra-id, service-name}` set

Reply:
gotcha, that works!